app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt

```diff
@@ -128,13 +139,20 @@ class ItineraryCreateActivity : BaseMapActivity(), OnMapReadyCallback,
             missionBuilder = ParallelogramMappingMissionService(droneService)
             missionDrawer = MapboxMissionDrawer(mapView, mapboxMap, style)
 
+            areaBuilderDrawer = MapboxAreaBuilderDrawer(mapView, mapboxMap, style)
+
             // Area - Need to be the last Drawer instanciated to allow draggable vertex
             areaBuilder = ParallelogramBuilder()
+            areaBuilder.onAreaChanged.add { onMissionSettingModified() }
             areaBuilder.onVerticesChanged.add { areaBuilderDrawer.draw(areaBuilder) }
             areaBuilder.onVerticesChanged.add { deleteButton.isEnabled = true }
+            for(vertex in initialArea){
+                areaBuilder.addVertex(vertex)
+            }
```

Comment:
This could be simplified :

```suggestion
            initialArea.forEach { areaBuilder.addVertex(it) }
```